src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/LearnReadOrientationModel.java

```diff
@@ -72,25 +80,22 @@
 
     @Override
     protected void onStartup(){
-        final MetricsFile<?, Integer> referenceSiteMetrics = readMetricsFile(refHistogramTable);
-        refHistograms = referenceSiteMetrics.getAllHistograms();
+        refHistograms = sumHistogramsFromFiles(refHistogramFiles);
```

Comment:
You should validate that the input lists all have the same length.  If there's some check to make sure the headers' orders correspond that would be good too.

Reply:
Validate input list size, done

Reply:
I added a check for headers of the histograms, and `TableReader` will ensure that that `altSiteRecords` are formatted properly.